tests/unit/legacy-0-6-x/test-module-for-component-test.js

```diff
@@ -389,6 +389,9 @@ moduleForComponent('Component Integration Tests', {
   integration: true,
   beforeSetup() {
     setResolverRegistry({
+      // registered manually to ensure template-only-glimmer-components tests pass
+      'component:my-component': Component,
```

Comment:
I don't understand, why is this needed?

Reply:
@Turbo87  Without this, the component in use is a template only component which (when the new optional feature is enabled) means that the behaviors being tested are a bit different.

Specifically the `{{name}}` in this components template will literally mean nothing because there is no `this` and `{{name}}` is implicitly `{{this.name}}`. The alternate solution to what I did here is to update the template to use `{{@name}}` instead, but that will _only_ work when running against Ember 3.1+ with the optional feature enabled.

Reply:
but that means it is only necessary in this specific scenario, right? I think we should probably only add it in this scenario then 🤔

Reply:
That's fine and I don't have a problem with doing that, but it doesn't really change anything. In the other scenarios an `Ember.Component` is already being created to back the template, by registering it here all the time we just make the behavior consistent across all of our versions and scenarios.

Reply:
my understanding was that before, we were implicitly testing that "old-style" template-only components work correctly too

Reply:
True, though there really is no difference prior to the new optional feature. Ember always used either the component JS file that you provided or `Ember.Component`. 

Either way, I'll update to keep things being tested in roughly the same manner...